### ISSUE TITLE
RCS1196 doesn't trigger when a cast expression(s) are present

### ISSUE THREAD

**Issue by IanKemp** [JosefPihrt/Roslynator#197]
Consider the following program, which demonstrates an instance when RCS1196 triggers correctly, and one where it incorrectly does not:

```csharp
static void Main(string[] args)
{
    var ayeEnum = A.X;
    var ayeInt = (int)ayeEnum;
    // RCS1196
    var b1 = Mapper.ToB(ayeEnum);
    // no RCS1196
    var b2 = Mapper.ToB((A)ayeInt);
    // previous line should be suggested to be rewritten as:
    var b3 = ((A)ayeInt).ToB();
}

enum A
{
    X,
    Y,
    Z,
}

enum B
{
    X,
    Y,
    Z,
}

static class Mapper
{
    public static A ToA(this B bee)
    {
        switch (bee)
        {
            case B.X: return A.X;
            case B.Y: return A.Y;
            case B.Z: return A.Z;
            default: throw new InvalidOperationException();
        }
    }

    public static B ToB(this A aye)
    {
        switch (aye)
        {
            case A.X: return B.X;
            case A.Y: return B.Y;
            case A.Z: return B.Z;
            default: throw new InvalidOperationException();
        }
    }
}
```
